test/functional/feature_evm_transferdomain.py

```diff
@@ -995,6 +995,122 @@ def invalid_transfer_evm_dvm_after_evm_tx(self):
+            "0x0000000000000000000000000000000000000000"
+        )
+        self.nodes[0].utxostoaccount({self.address: "200@DFI"})
+        self.nodes[0].generate(1)
+
+        self.address_erc55 = self.nodes[0].addressmap(self.address, 1)["format"][
+            "erc55"
+        ]
+
+        transfer_domain(
+            self.nodes[0], self.address, self.address_erc55, "100@DFI", 2, 3
+        )
+        self.nodes[0].generate(1)
+
+        self.nodes[0].eth_sendTransaction(
+            {
+                "nonce": self.nodes[0].w3.to_hex(
+                    self.nodes[0].w3.eth.get_transaction_count(self.address_erc55)
+                ),
+                "from": self.address_erc55,
+                "to": burn_address,
+                "value": "0x1",
+                "gas": "0x100000",
+                "gasPrice": "0x4e3b29200",
+            }
+        )
+        self.nodes[0].generate(1)
+
+        balance = Decimal(self.nodes[0].w3.eth.get_balance(self.address_erc55))
+        burn_balance = Decimal(self.nodes[0].w3.eth.get_balance(burn_address))
+        nonce = self.nodes[0].w3.eth.get_transaction_count(self.address_erc55)
+        sender_nonce = self.nodes[0].w3.eth.get_transaction_count(self.address_erc55)
+
+        self.nodes[0].transferdomain(
+            [
+                {
+                    "src": {"address": self.address, "amount": "100@DFI", "domain": 2},
+                    "dst": {
+                        "address": self.address_erc55,
+                        "amount": "100@DFI",
+                        "domain": 3,
+                    },
+                    "nonce": nonce,
+                }
+            ]
+        )
+        self.nodes[0].eth_sendTransaction(
+            {
+                "nonce": self.nodes[0].w3.to_hex(nonce),
+                "from": self.address_erc55,
+                "to": "0x0000000000000000000000000000000000000000",
+                "value": "0x1",
+                "gas": "0x100000",
+                "gasPrice": "0x4e3b29200",
+            }
+        )
+        self.nodes[0].generate(1)
+
+        balance_after = Decimal(self.nodes[0].w3.eth.get_balance(self.address_erc55))
+        burn_balance_after = Decimal(self.nodes[0].w3.eth.get_balance(burn_address))
+        sender_nonce_after = self.nodes[0].w3.eth.get_transaction_count(
+            self.address_erc55
+        )
+
+        # evmtx should take precedence, transferdomain should be discarded
+        assert_equal(
+            sender_nonce + 1, sender_nonce_after
+        )  # evm tx will increment account nonce
+        assert_equal(
+            burn_balance_after, burn_balance + Decimal(1)
+        )  # null address should have more balance
+        assert balance_after < balance  # sender should have less balance
+
+    def should_find_empty_nonce(self):
+        self.nodes[0].clearmempool()
+        nonce = self.nodes[0].w3.eth.get_transaction_count(self.address_erc55)
+
+        evm_nonces = [nonce, nonce + 2, nonce + 3]
```

Comment:
Can we do the exact same test but test the auto nonce for evmtx RPC and eth_sendTransaction RPC?